cool-core/src/main/java/com/nus/cool/core/cohort/refactor/olapSelect/olapSelectionLayout.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.nus.cool.core.cohort.refactor.olapSelect;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.nus.cool.core.cohort.refactor.filter.Filter;
+import com.nus.cool.core.cohort.refactor.filter.FilterLayout;
+import com.nus.cool.core.cohort.refactor.filter.FilterType;
+import com.nus.cool.core.io.readstore.MetaChunkRS;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import lombok.Data;
+
+@Data
+public class olapSelectionLayout {
+
+  // inner data type for selection
+  public enum SelectionType {
+    and, or, filter
+  }
+
+  // what type of selection
+  @JsonProperty("filter_type")
+  private SelectionType type;
+
+  // which dimension to use
+  @JsonProperty("fieldSchema")
+  private String dimension;
+
+  // dimension type
+  @JsonProperty("type")
+  private FilterType dimension_type;
+
+  // values accepted in this field
+  @JsonProperty("acceptValue")
+  private List<String> values;
+
+  // sub-operation
+  @JsonProperty("fields")
+  private List<olapSelectionLayout> fields = new ArrayList<>();
```

Comment:
Circular reference ? This class is wired.

 According to the logic in olapSelectionLayout, it seems that there is a nested selectionLayout.
SelectionLayout contains fields which is a list of SelectionLayout.

According to the logic in olapSelector, the select method only support one-level process.

Maybe we should discuss about logic in this part and give a more clear implementation.

Reply:
Updated in https://github.com/COOL-cohort/COOL/issues/122